### PR TITLE
[mod_event_socket] fix possible memory-leak in event_sink_function()

### PR DESCRIPTION
**Pull request by bit4bit**

### CODE REVIEW

File: src/mod/event_handlers/mod_event_socket/mod_event_socket.c [hunk 1]
```diff
@@ -1109,13 +1109,15 @@ SWITCH_STANDARD_API(event_sink_function)
 					switch_xml_free(xml);
 				} else {
 					stream->write_function(stream, "<data><reply type=\"error\">XML Render Error</reply></data>\n");
-					break;
+					goto endloop;
```
Comment (andywolk) [inline src/mod/event_handlers/mod_event_socket/mod_event_socket.c:1112]: That continues the loop, does not break. Why?

Comment (bit4bit) [reply src/mod/event_handlers/mod_event_socket/mod_event_socket.c:1112]: sorry i'm wrong

i dont see at  https://github.com/signalwire/freeswitch/blob/da8703b2151ebdbb9331a10c5814c016cdd282c4/src/mod/event_handlers/mod_event_socket/mod_event_socket.c#L1137

thanks for your time

**Comment by bit4bit** [comment]
sorry i'm wrong

i do not look at https://github.com/signalwire/freeswitch/blob/da8703b2151ebdbb9331a10c5814c016cdd282c4/src/mod/event_handlers/mod_event_socket/mod_event_socket.c#L1137

thanks for your time

**Comment by andywolk** [comment]
May be you are not wrong, May be call `switch_event_destroy(&pevent);` before `break` ?

**Comment by andywolk** [comment]
@bit4bit

**Comment by bit4bit** [comment]
hi @andywolk sorry my mistake

https://github.com/signalwire/freeswitch/blob/c4115537e0c1c24abad011c102e25041d18bc7b4/src/mod/event_handlers/mod_event_socket/mod_event_socket.c#L1134-L1136

**Pull request closed without merging**